meteor/client/ui/Settings/Studio/Routings.tsx

```diff
@@ -12,757 +10,1031 @@ import {
+					options={getDropdownInputOptions(StudioRouteType)}
+				>
+					{(value, handleUpdate, options) => {
+						if (!rawRoute.mappedLayer) {
+							return <span className="mls">REMAP</span>
+						} else {
+							return (
+								<DropdownInputControl
+									classNames="input text-input input-l"
+									options={options}
+									value={value}
+									handleUpdate={handleUpdate}
+								/>
+							)
+						}
+					}}
+				</LabelAndOverridesForDropdown>
+
+				<LabelAndOverridesForDropdown
+					label={t('Device Type')}
+					item={route}
+					itemKey={'deviceType'}
+					opPrefix={route.id}
+					overrideHelper={tableOverrideHelper}
+					options={getDropdownInputOptions(TSR.DeviceType)}
+				>
+					{(value, handleUpdate, options) => {
+						if (rawRoute.routeType === StudioRouteType.REROUTE && rawRoute.mappedLayer) {
+							return deviceTypeFromMappedLayer !== undefined ? (
+								<span className="mls">{TSR.DeviceType[deviceTypeFromMappedLayer]}</span>
+							) : (
+								<span className="mls dimmed">{t('Original Layer not found')}</span>
+							)
+						} else {
+							return (
+								<DropdownInputControl
+									classNames="input text-input input-l"
+									options={options}
+									value={value}
+									handleUpdate={handleUpdate}
+								/>
+							)
+						}
+					}}
+				</LabelAndOverridesForDropdown>
+
+				{mappingTypeOptions.length > 0 && (
+					<LabelAndOverridesForDropdown<any> // Deep key is not allowed, but is fine for now
+						label={t('Mapping Type')}
+						item={route}
+						itemKey={'remapping.options.mappingType'}
+						opPrefix={route.id}
+						overrideHelper={tableOverrideHelper}
+						options={mappingTypeOptions}
+					>
+						{(value, handleUpdate, options) => (
+							<DropdownInputControl
+								classNames="input text-input input-l"
+								options={options}
+								value={value}
+								handleUpdate={handleUpdate}
+							/>
+						)}
+					</LabelAndOverridesForDropdown>
+				)}
+				{rawRoute.routeType === StudioRouteType.REMAP ||
+				(routeDeviceType !== undefined && rawRoute.remapping !== undefined) ? (
+					<>
+						<LabelAndOverrides<any> // Deep key is not allowed, but is fine for now
+							label={t('Device ID')}
+							item={route}
+							itemKey={'remapping.deviceId'}
+							opPrefix={route.id}
+							overrideHelper={tableOverrideHelper}
+						>
+							{(value, handleUpdate) => (
+								<div>
+									<CheckboxControl
+										classNames="input"
+										title={t('Enable/Disable route set override')}
+										value={value !== undefined}
+										handleUpdate={() => handleUpdate(undefined)}
+									/>
+									<TextInputControl
+										modifiedClassName="bghl"
+										classNames="input text-input input-l"
+										value={value}
+										handleUpdate={handleUpdate}
+									/>
+								</div>
+							)}
+						</LabelAndOverrides>
+
+						{/** TODO: this needs the same checkbox to enable/disable as above */}
```

Comment:
I think this needs addressing, otherwise it is possible to add a route to the routeset which will for example change the caspar layer it is addressing, but then the ui won't let you clear that.